File: server/datasets/python_source-code/Multiple-D1/organic_10.py
"""
Quick Sort - Organic Submission #10
Encapsulated Object-Oriented QuickSorter class.
"""

class QuickSorter:
    def __init__(self, data):
        self.data = list(data)

    def _partition(self, low, high):
        pivot = self.data[high]
        i = low - 1
        for j in range(low, high):
            if self.data[j] <= pivot:
                i += 1
                self.data[i], self.data[j] = self.data[j], self.data[i]
        self.data[i + 1], self.data[high] = self.data[high], self.data[i + 1]
        return i + 1

    def _sort(self, low, high):
        if low < high:
            pi = self._partition(low, high)
            self._sort(low, pi - 1)
            self._sort(pi + 1, high)

    def sort(self):
        self._sort(0, len(self.data) - 1)
        return self.data

if __name__ == "__main__":
    sorter = QuickSorter([5, 2, 9, 1, 7, 6, 3])
    print(sorter.sort())

File: server/datasets/java_source-code/Multiple-D1/organic_10.java
/*
 * Quick Sort - Organic Submission #10
 * Encapsulated Object-Oriented QuickSorter class.
 */

class QuickSorter {
    private int[] data;

    public QuickSorter(int[] input) {
        this.data = input.clone();
    }

    private int partition(int low, int high) {
        int pivot = this.data[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (this.data[j] <= pivot) {
                i++;
                int temp = this.data[i];
                this.data[i] = this.data[j];
                this.data[j] = temp;
            }
        }
        int temp = this.data[i + 1];
        this.data[i + 1] = this.data[high];
        this.data[high] = temp;
        return i + 1;
    }

    private void sortHelper(int low, int high) {
        if (low < high) {
            int pi = partition(low, high);
            sortHelper(low, pi - 1);
            sortHelper(pi + 1, high);
        }
    }

    public int[] sort() {
        sortHelper(0, this.data.length - 1);
        return this.data;
    }

    public static void main(String[] args) {
        QuickSorter sorter = new QuickSorter(new int[]{5, 2, 9, 1, 7, 6, 3});
        System.out.println(java.util.Arrays.toString(sorter.sort()));
    }
}
